node/src/chain_specs/manta/mainnet_genesis.rs

```diff
@@ -0,0 +1,194 @@
+use super::*;
+use hex_literal::hex;
+use sc_network::config::MultiaddrWithPeerId;
+use sc_telemetry::TelemetryEndpoints;
+use sp_core::crypto::UncheckedInto;
+#[derive(Clone)]
+struct Collator {
+    acc: AccountId,
+    nodeid: MultiaddrWithPeerId,
+    keys: SessionKeys,
+}
+impl Collator {
+    fn new(acc: AccountId, nodeid: MultiaddrWithPeerId, keys: SessionKeys) -> Collator {
+        Self { acc, nodeid, keys }
+    }
+}
+
+fn manta_mainnet_genesis(genesis_collators: Vec<Collator>) -> GenesisConfig {
+    const TOTAL_ISSUANCE: Balance = 1_000_000_000 * MANTA;
+    const INITIAL_CONTROLLER_BALANCE: Balance = 5_000 * MANTA;
+    const INITIAL_COLLATOR_BALANCE: Balance = 100 * MANTA;
+    let root_key: AccountId =
+        hex!("fb1818aca757d6bc9aa65c3a94f5a39ff1edbd92e47311c91ace601676168703").into(); // MULTISIG // ss58: 16gEBH3Nq4mtWGg9oFBGveSSoUDcKRABwAq9KESiWKEwb1vQ
+    let endowments = vec![
+        (
+            hex!("4a5d27062c9d0b5747566139c683c17a0e6da8c440c41dfa01725ba664af6977").into(),
+            INITIAL_CONTROLLER_BALANCE,
+        ), // CONTROLLER 1 ss58 12gWDW1z7T1AN1Vm86RcvRdSmyjtJ1Mv4LHmYvmq9LJG8czh
+        (
+            hex!("a4c2839433ae05aa48e615e5a6917cc98e4a2886272f2519347f2be7c0c6853d").into(),
+            INITIAL_CONTROLLER_BALANCE,
+        ), // CONTROLLER 2 ss58 14j2dssnCyQD2xWezTvZmbZPSKDBrwKcAcBtXrE6XgeGfzGx
+        (
+            hex!("3e31eaf98e0f6dd9b088c095757fac00dc9e3fe6af29e42caea94f7d76dd3444").into(),
+            INITIAL_CONTROLLER_BALANCE,
+        ), // CONTROLLER 3 ss58 12QYoFyrrEomSq4tMPxVxnWvPwVVKQHUpDWZuDxcg5Gv3c9E
+        (
+            hex!("0619e4018531059b677d70b476b72431d3e88fe71d2c27bba5d945975e824956").into(),
+            INITIAL_CONTROLLER_BALANCE,
+        ), // COLLATOR 1
+        (
+            hex!("0202d1760e33f60f977fcff88b611cb95e66d0b3b74c12ac6c8037b8b9764a31").into(),
+            INITIAL_COLLATOR_BALANCE,
+        ), // COLLATOR 2
+        (
+            hex!("4ece9156edd66af83b0c6afe17878844d873a08fd05bd9dbc17f9712814e9952").into(),
+            INITIAL_COLLATOR_BALANCE,
+        ), // COLLATOR 3
+        (
+            hex!("7a5cef43508640b6839c37a41272dec83dda4df28503bd87436c197d213f692c").into(),
+            INITIAL_COLLATOR_BALANCE,
+        ), // COLLATOR 4
+        (
+            hex!("4039a894994d326e32801a0fc19a267383b6231e74bc9c3fd1021d949d849406").into(),
+            INITIAL_COLLATOR_BALANCE,
+        ), // COLLATOR 5
+        (
+            root_key.clone(),
+            TOTAL_ISSUANCE - 4 * INITIAL_CONTROLLER_BALANCE - 4 * INITIAL_COLLATOR_BALANCE,
```

Comment:
Are we going to assign all rest of tokens to root?

Reply:
yes